src/lib.rs

```diff
@@ -562,44 +563,68 @@ where
     ) {
         let mandatory_parameter_count = parameters
             .iter()
-            .filter(|p| match p {
-                Parameter::Mandatory { .. } => true,
-                _ => false,
-            })
+            .filter(|p| matches!(p, Parameter::Mandatory { .. },))
             .count();
         let positional_parameter_count = parameters
             .iter()
-            .filter(|p| match p {
-                Parameter::Mandatory { .. } => true,
-                Parameter::Optional { .. } => true,
-                _ => false,
-            })
+            .filter(|p| matches!(p, Parameter::Mandatory { .. } | Parameter::Optional { .. }))
             .count();
         if command.len() >= item.command.len() {
             // Maybe arguments
             let mut argument_buffer: [&str; 16] = [""; 16];
             let mut argument_count = 0;
             let mut positional_arguments = 0;
-            for (slot, arg) in argument_buffer
-                .iter_mut()
-                .zip(command[item.command.len()..].split_whitespace())
-            {
-                *slot = arg;
-                argument_count += 1;
-                if arg.starts_with("--") {
+            // buffer of characters and lengths
+            let mut buf = [([0u8; 32], 0_usize); 16];
```

Comment:
32 x 16 bytes is quite a big increase of memory usage compared to what we had before.

If this remains it should be a const generic.

Reply:
Or it might be better to de-escape the buffer in place. Any de-escaped string must be the same length or shorter than the input string, and so that should always fit into the buffer that's available.